iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -197,9 +197,9 @@ PublisherPortUserType::MemberType_t* PoshRuntime::getMiddlewarePublisher(const c
 
     IpcMessage sendBuffer;
     sendBuffer << IpcMessageTypeToString(IpcMessageType::CREATE_PUBLISHER) << m_appName
-               << static_cast<cxx::Serialization>(service).toString() << std::to_string(options.historyCapacity)
-               << options.nodeName << std::to_string(options.offerOnCreate)
-               << std::to_string(static_cast<uint8_t>(options.subscriberTooSlowPolicy))
+               << static_cast<cxx::Serialization>(service).toString() << cxx::convert::toString(options.historyCapacity)
+               << options.nodeName << cxx::convert::toString(options.offerOnCreate)
+               << cxx::convert::toString(static_cast<uint16_t>(options.subscriberTooSlowPolicy))
```

Comment:
using static_cast<uint8_t> here returns '/100' from toString()

Reply:
This is a bug in the `cxx::convert::toString` function. It uses `stringstream` to convert to a string and a `uint8_t` is actually a `unsigned char` which will not be converted since it is already  `char`. Could you add a specialization for `uint8_t` and `int8_t` to `cxx::convert::toString` which does the cast to a `uint16_t`/`int16_t` and then calls the generic `cxx::convert::toString`? Tests would also be necessary.